### DISCUSSION TITLE
How to update TestCaseId in reportportal.io executing TC from Specflow

### DISCUSSION THREAD

**Discussion by sureshp191** [reportportal/reportportal#1599] [Q&A] [answered]
I have configured my Specflow+Xunit project with reportPortal.io I want to set "TestCaseId:" in report portal for testItem details popUp When I tried to set as tag in .feature file it flows in reportPorta.io as Attribute. Seeing as below in featurefile.

![image](https://user-images.githubusercontent.com/26474888/144152540-29d6b623-3360-44bc-bfdb-ebe220e382b0.png)

in reportPortal.io it appears as below as Attribute not as "TestCaseId".

![image](https://user-images.githubusercontent.com/26474888/144152577-de1199de-e25a-4abb-85ca-33abfcfd22ca.png)

**Comment by evjlobanova** [comment]
@nvborisenko Could you please share an example of how to report test case id?

**Comment by nvborisenko** [comment, marked as answer]
Example how to modify reporting: https://github.com/reportportal/example-net-specflow/blob/master/Example.SpecFlow/Hooks/Hooks1.cs

In your case subscribe on `ReportPortalAddin.BeforeScenarioStarted` event

```csharp
public static void BeforeTestRunPart()
{
  ReportPortalAddin.BeforeScenarioStarted += ReportPortalAddin_BeforeScenarioStarted;
}
```

And then in handler set `TestCaseId`
```csharp
private static void ReportPortalAddin_BeforeScenarioStarted(object sender, TestItemStartedEventArgs e)
{
  e.StartTestItemRequest.TestCaseId = "whatever you want";

  // e.ScenarioContext.ScenarioInfo.Tags helps you to get the list of assigned tags
}
```

**Reply by sureshp191** [reply]
Thank you it worked.

**Reply by piyush7870** [reply]
@nvborisenko is it possible to achieve same with xunit framework?(not specflow)

**Reply by nvborisenko** [reply]
Yes, but in slightly different way. Runner and tests are living in different processes. So, you can create any `My.ReportPortal.SuperExtension.dll` (ReportPortal entry is mandatary in assembly name). This dll should be located near `ReportPortal.XUnitReporter.dll` to let the agent pick up your dll.

Create class which implements `ReportPortal.Shared.Extensibility.IReportEventsObserver` from https://www.nuget.org/packages/ReportPortal.Shared nuget package, where you can subscribe on `OnBeforeTestStarting` event, and modify outgoing request to report portal.

Extensibility mechanism of agents is not documented yet, you can rely on internal tests:  https://github.com/reportportal/commons-net/blob/0164aac7880bb8f3ce7393a8d9346136efebf583/test/ReportPortal.Shared.Tests/Extensibility/ReportEvents/ReportEventsFixture.cs#L224
